controllers/azuremachine_controller.go

```diff
@@ -235,29 +252,35 @@ func (r *AzureMachineReconciler) reconcileNormal(ctx context.Context, machineSco
 	switch vm.State {
 	case infrav1.VMStateSucceeded:
 		machineScope.V(2).Info("VM is running", "id", *machineScope.GetVMID())
+		conditions.MarkTrue(machineScope.AzureMachine, infrav1.VMReadyCondition)
 		machineScope.SetReady()
 	case infrav1.VMStateCreating:
 		machineScope.V(2).Info("VM is creating", "id", *machineScope.GetVMID())
+		conditions.MarkFalse(machineScope.AzureMachine, infrav1.VMReadyCondition, infrav1.VMNotReadyReason, clusterv1.ConditionSeverityWarning, "")
```

Comment:
I'm not sure `VMStateCreating` should be a `ConditionSeverityWarning`, because I assume this is part of the normal provisioning workflow. What about using `ConditionSeverityInfo` instead?
Should we apply the same consideration to `VMStateUpdating`?

Reply:
makes sense